### ISSUE TITLE
Break out the web related code into a standalone package

### ISSUE THREAD

**Issue by mattyjones** [phantomSecrets/phantomSecrets#18]
The web sever should be a standalone package for the sake of portability and code organization.

**Comment by codeEmitter** [comment]
Could we use docker here?  Not sure we want to tie the tool to the docker platform, but it would provide a lot of benefit.

**Comment by mattyjones** [comment]
I agree docker could be used but how would we handle the communications between them. We expose port <port> to hit the webserver but how do we get the data into the container easily unless we run the whole thing in docker, which may cause resource problems. I know running Ansible last night caused by MBP to cry for a solid 3-4 minutes

<img width="1680" alt="Screen Shot 2020-06-05 at 8 32 16 PM" src="https://user-images.githubusercontent.com/672940/83949348-408a8c00-a7f1-11ea-811e-81ddd6d427ff.png">

**Comment by codeEmitter** [comment]
Well, that's just it.  The api that's currently served from the binary would have to be moved out as well.  I could see us breaking this up in a way that could either be served from docker/k8s, or not depending on purpose.  Main project contains all the packages.  Maybe another project could orchestrate the packages however we like.  Standalone binary deployment as is what happens now, or a containerized platform of some kind for more robust requirements.

Just thinking out loud and exploring ideas.

**Comment by codeEmitter** [comment]
I still think there's value in a standalone binary deployment, but as you scale out your scans you definitely need more flexibility in the architecture.  From my experience that is.

**Comment by codeEmitter** [comment]
I don't think docker will kill local performance.  It's a very different thing than what ansible does.  In any case, I think having a solid core set of packages is the main object still.  Assemblance of the packages into backend services, web server(s), CLI implementation could be done separately perhaps, either in separate projects or in the same project.  If we do our jobs right letting the tests drive the architectural design of the packaging seams, then we'll land in the right spot to do almost anything we want with the packages.

**Comment by mattyjones** [comment]
I agree which is why I started on tests first. I am also adding doc strings, code comments, and any error handling needed as I go through each file. This is going to take a while but will be well worth it in the long run. My thinking was to simply go package by package with the testing. Common was the smallest and easiest so I went there first.
